pkg/webservice/handlers.go

```diff
@@ -188,6 +211,31 @@ func getClusterJSON(name string) *dao.ClusterDAOInfo {
 	return clusterInfo
 }
 
+func getClusterUtilJSON(partition *cache.PartitionInfo, name string) *dao.ClusterUtilDAOInfo {
+	var total, used resources.Quantity
+	var percent int64
+	total, ok := partition.GetTotalPartitionResource().Resources[name]
+	if !ok {
+		total = 0
+	}
+	used, ok = partition.Root.GetAllocatedResource().Resources[name]
+	if !ok {
+		used = 0
+	}
+	if total < used || total == 0 {
+		percent = int64(0)
+	} else {
+		percent = ((int64(used) * int64(100)) / int64(total))
```

Comment:
This needs some protection against integer overflow. Both positive and negative. For example if If used = maxInt64, available is 10 then we will have an overflow here. You can find an example in the resources.go fileÉ https://github.com/apache/incubator-yunikorn-core/blob/master/pkg/common/resources/resources.go#L264-L279

Reply:
Also I think we need to cover the edge cases as well with unit test: nil values, zero values, inf values

Reply:
In resources.go file there is already a method for calculating absolute usage: func CalculateAbsUsedCapacity(capacity, used *Resource) *Resource please use that one if possible.

Reply:
done